crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -2313,6 +2313,22 @@ impl ForkState {
                 // A dependency from the root package or requirements.txt.
                 debug!("Adding direct dependency: {package}{version}");
 
+                let name = package.name_no_root().unwrap();
+
+                // Catch cases where we pass a package once by name with extras and then once as
+                // URL for the specific distribution.
+                has_url = has_url
+                    || dependencies
+                        .iter()
+                        .filter(|other_dep| *other_dep != dependency)
+                        .filter(|other_dep| {
+                            other_dep
+                                .package
+                                .name()
+                                .is_some_and(|other_name| other_name == name)
+                        })
+                        .any(|other_dep| other_dep.url.is_some());
```

Comment:
This is quadratic, right?

Reply:
Forgot to add the comment to the PR: This only runs for direct deps, so the quadratic loop runs about once, and in the number of direct dependencies.